itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/txn/compactor/TestMaterializedViewRebuild.java

```diff
@@ -97,7 +91,7 @@ public void testWhenMajorCompactionThenIncrementalMVRebuildIsStillAvailable() th
     txnHandler.cleanTxnToWriteIdTable();
 
     List<String> result = execSelectAndDumpData("explain cbo alter materialized view " + MV1 + " rebuild", driver, "");
-    Assert.assertEquals(INCREMENTAL_REBUILD_PLAN, result);
+    Assert.assertEquals(FULL_REBUILD_PLAN, result);
```

Comment:
We search for update/delete operations in the `COMPLETED_TXN_COMPONENTS` affected source tables at MV rebuild. Records are deleted from this table at compaction. So after compaction we can not confirm whether there were any deletes of any of the source tables any longer. It is relevant since executing an incremental rebuild plan which expects insert operations in all source table only in case there were deletes leads to data corruption in the refreshed view.

The second rebuild can be an incremental since the first rebuild resets the source tables snapshot to a fresh one and txn data of operations done since that first rebuild still exists in `COMPLETED_TXN_COMPONENTS`.

Reply:
Thanks for the explanation.  Makes sense that since the entries are deleted from the COMPLETED_TXN_COMPONENTS table at compaction, there's no way to check if only insert operation was done in a full ACID table.